### ISSUE TITLE
`yas/expand` expands but `yas-expand` does not.

### ISSUE THREAD

**Issue by phfrohring** [capitaomorte/yasnippet#656]
Hello !

For some reason, yasnippet stopped working when using org-mode.

I'm editing something and have my cursor like so: 
```
pro▯
```

I hit `tab` then: nothing.

I hit `C-c h k <tab>`, and I get:
![screen shot 2016-01-29 at 15 22 59](https://cloud.githubusercontent.com/assets/10885649/12678844/21a5026a-c6a1-11e5-88a2-b0dbb308b515.png)

I go back to previous state:
```
pro▯
```

then: `M-x yas/expand` then:
```
(pro b2ee)
```

Problem: `yas-expand ≠ yas/expand` ? or maybe `yasnippet` and `org-mode` don't play well together?

I tried http://orgmode.org/manual/Conflicts.html but no result: a strange error saying that `yas-minor-mode` should be properly initialized...

**Comment by npostavs** [comment]
Can you provide a recipe to show the problem starting from `emacs -Q`? See https://github.com/capitaomorte/yasnippet#important-note-regarding-bug-reporting

If I do `emacs -Q -L . -l yasnippet -f yas-global-mode` and open an org-mode buffer, I can expand snippets, no problem.

> I tried http://orgmode.org/manual/Conflicts.html but no result: a strange error saying that yas-minor-mode should be properly initialized...

The advice here is out of date, `yas/trigger-key` no longer does anything.

**Comment by phfrohring** [comment]
I updated to the last version of yasnippet and :tada: : cannot reproduce the bug anymore!
So, I close this issue.
Thx for being so reactive!

**Comment by phfrohring** [comment]
Well, I think I understood.  It's because of the ambiguity of `<tab>` in org-mode.
let's say that one have a snippet: `x → XXX` so we have this sequence in a file: 
1. text := `x▯`
2. press `<tab>`
3. text := `XXX▯`

if org-mode & cursor on multiline list:

1 text :=
```
  - x▯ blablabla
    blablabla
```

2 press `<tab>`

3 Expect:
```
  - XXX▯ blablabla
    blablabla
```
But got:
```
  - x▯ blablabla...    
```

So I guess it has to do with the way `<tab>` is interpreted: "What do you mean Emacs user? org-mode tab or yas/expand?"

If single line list item then it works.

So I guess the disambiguation works like: "org-mode tab did not find anything to do then let other interpretation run. Ho, goolies! yas/expand found something! And here you go, have fun with your substitutions."

**Comment by npostavs** [comment]
Works for me. I got the expected output.

---

> So I guess the disambiguation works like: "org-mode tab did not find anything to do then let other interpretation run. Ho, goolies! yas/expand found something! And here you go, have fun with your substitutions."

Actually, `yas-expand` gets first go and then decides what to do based on `yas-fallback-behavior` if there is no snippet. By default it runs whatever would have run without yasnippet. The docstring should say so if you do `<f1> k <tab>`.

```
<tab> runs the command yas-expand, which is an alias for
`yas-expand-from-trigger-key' in `yasnippet.el'.

It is bound to TAB, <tab>, <menu-bar> <YASnippet> <Expand trigger>.

(yas-expand &optional FIELD)

Expand a snippet before point. If no snippet
expansion is possible, call command `org-cycle'.

Optional argument FIELD is for non-interactive use and is an
object satisfying `yas--field-p' to restrict the expansion to.
```

**Comment by phfrohring** [comment]
... well then, I don't know why sometime it works and sometime it doesn't. 
If I come back here, it will be with reproducible behavior.
Thx for answering anyway! :+1:

**Comment by zzhjerry** [comment]
I met the same problem. I'm in js2-mode and want to expand `/**` to
```
/**
 *
 */
```
but `yas-expand` can't expand the snippet. While `yas/expand` did

**Comment by npostavs** [comment]
@zzhjerry Can you reproduce this from [`emacs -Q`][emacs-Q]? I'm in the process of changing the way TAB binding and fallback works, so it might get fixed by that anyway, see #760.

[emacs-Q]: http://www.lunaryorn.com/2015/11/29/reproduce-bugs-in-emacs-Q.html
